ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/ApplicationEngineEnvironmentJvm.kt

```diff
@@ -72,8 +72,69 @@ public actual class ApplicationEngineEnvironmentBuilder {
      */
     public actual fun build(builder: ApplicationEngineEnvironmentBuilder.() -> Unit): ApplicationEngineEnvironment {
         builder(this)
-        return ApplicationEngineEnvironmentReloading(
-            classLoader, log, config, connectors, modules, watchPaths, parentCoroutineContext, rootPath, developmentMode
+        if (developmentMode) {
+            val result = ApplicationEngineEnvironmentReloading(
+                classLoader,
+                log,
+                config,
+                connectors,
+                modules,
+                watchPaths,
+                parentCoroutineContext,
+                rootPath,
+                developmentMode
+            )
+
+            if (result != null) return result
+            val message = "Failed to initialize reloading environment. " +
```

Comment:
why not to throw exception here, as anyway it's `dev` mode?
And in this case also, f.e. autoreload can be explicitly disabled if `watchPaths` is empty or via some flag?

Reply:
`dev` mode is not only auto-reloading, it's stack traces. 

Yep, let's make it disabled on empty watchPaths